lib/src/request.dart

```diff
@@ -113,10 +113,10 @@ class Request extends BaseRequest {
           'content-type "application/x-www-form-urlencoded".');
     }
 
-    return Uri.splitQueryString(body, encoding: encoding);
+    return splitQueryString(body, encoding: encoding);
```

Comment:
> Here is example code of what a MultiMap would look like (if I coded it)
> https://github.com/rwrozelle/multimap
> 
> Is this what you are looking for?  I've played around with using it in local copies of the http and http_server package and it definitely plays nicely with only minimal updates to existing code and backward compatible.  I decided on using a parent/child component template because it was the least amount of work and I didn't heavily research the existing Mixin classes, so might need to add some "with" stuff on the class definition?

The behavior here looks more or less right. We can iterate on the details of the design in a separate CL.

> I'm having a heck of a time compiling the SDK, I need to build a new VM on my laptop to get everything lined up and don't have the time until next year, so I can't put together example of code integrated into Collections package.

Good news: you don't have to add it to the SDK! The [`collection` pub package](https://github.com/dart-lang/collection) is the place for new collection types these days.